xonsh/tools.py

```diff
@@ -343,6 +343,16 @@ def command_not_found(cmd):
     return s
 
 
+def _compute_known_commands(env, aliases):
+    list_ = []
```

Comment:
A more common pattern for this would be `list_ = list(aliases)` (Python 2+) or `list_ = [*aliases]` (Python 3.5+).